packages/kit/src/runtime/app/paths/server.js

```diff
@@ -13,19 +13,17 @@ export function resolve(id, params) {
 	const resolved = resolve_route(id, /** @type {Record<string, string>} */ (params));
 
 	if (relative) {
-		const { event, state } = get_request_store();
+		const store = try_get_request_store();
 
-		if (state.prerendering?.fallback) {
-			return resolved;
-		}
-
-		const segments = event.url.pathname.slice(base.length).split('/').slice(2);
-		const prefix = segments.map(() => '..').join('/') || '.';
+		if (store && !store.state.prerendering?.fallback) {
```

Comment:
Not sure if this the correct path (typed this on my phone) but this is how we can ensure remote functions don't get relative paths (though when called while server rendering it would still be relative inside remote functions; not sure if that is important)
```suggestion
		if (store && !store.event.isRemoteFunction && !store.state.prerendering?.fallback) {
```

Reply:
I think we really need something that works consistently wherever it's called from — maybe warning if it's used in a remote function when `paths.relative` is `true` or something like that. Opened #14537 for us to come back to